x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/actions_menu.test.tsx

```diff
@@ -121,18 +133,38 @@ describe('When using the Endpoint Details Actions Menu', () => {
   describe('and endpoint host is isolated', () => {
     beforeEach(() => setEndpointMetadataResponse(true));
 
-    it('should display Unisolate action', async () => {
-      await render();
-      expect(renderResult.getByTestId('unIsolateLink')).not.toBeNull();
+    describe('and user has unisolate privilege', () => {
+      it('should display Unisolate action', async () => {
+        await render();
+        expect(renderResult.getByTestId('unIsolateLink')).not.toBeNull();
+      });
+
+      it('should navigate via router when unisolate is clicked', async () => {
+        await render();
+        act(() => {
+          fireEvent.click(renderResult.getByTestId('unIsolateLink'));
+        });
+
+        expect(coreStart.application.navigateToApp).toHaveBeenCalled();
+      });
     });
 
-    it('should navigate via router when unisolate is clicked', async () => {
-      await render();
-      act(() => {
-        fireEvent.click(renderResult.getByTestId('unIsolateLink'));
+    describe('and user does not have unisolate privilege', () => {
+      beforeEach(() => {
+        (useUserPrivileges as jest.Mock).mockReturnValue({
+          ...initialUserPrivilegesState(),
+          endpointPrivileges: {
+            ...initialUserPrivilegesState().endpointPrivileges,
+            canIsolateHost: true,
+            canUnIsolateHost: false,
```

Comment:
Is this a valid condition?  Can a user have Isolate authz but NOT un-isolate?

Reply:
that's true, currently no. will update.